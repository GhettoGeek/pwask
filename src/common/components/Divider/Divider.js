import React from 'react'
import PropTypes from 'prop-types'
import Divider from '@material-ui/core/Divider'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles({
  divider: {
    height: 28,
    margin: 4,
  },
})

function EnhancedDivider({
  variant, orientation, light, absolute,
}) {
  const classes = useStyles()

  return (
    <Divider
      light={light}
      variant={variant}
      className={classes.divider}
      orientation={orientation}
      absolute={absolute}
    />
  )
}

EnhancedDivider.propTypes = {
  light: PropTypes.bool,
  variant: PropTypes.oneOf(['fullWidth', 'inset', 'middle']),
  orientation: PropTypes.oneOf(['vertical', 'horizontal']),
  absolute: PropTypes.bool,
}

EnhancedDivider.defaultProps = {
  light: false,
  variant: 'fullWidth',
  orientation: 'horizontal',
  absolute: false,
}

export default EnhancedDivider
