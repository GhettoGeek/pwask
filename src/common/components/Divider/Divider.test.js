import React from 'react'
import { shallow } from 'enzyme'
import Divider from './Divider'

describe('Divider', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(<Divider orientation="vertical" />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
