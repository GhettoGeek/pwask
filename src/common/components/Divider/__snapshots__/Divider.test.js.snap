// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly the component 1`] = `"<hr class=\\"MuiDivider-root makeStyles-divider-1 MuiDivider-vertical\\"/>"`;
